### DISCUSSION TITLE
Add ability to boot from loopback device images on bare metal

### DISCUSSION THREAD

**Discussion by bam80** [systemd/mkosi#3272] [Ideas]
Dracut has [`iso-scan/filename`](https://www.kernel.org/pub/linux/utils/boot/dracut/dracut.html#_iso_scan_filename) option to boot from `.ISO` image on a filesystem:
>  Mount all mountable devices and search for ISO pointed by the argument. When the ISO is found set it up as a loop device. Device containing this ISO image will stay mounted at /run/initramfs/isoscandev. Using iso-scan/filename with a Fedora/Red Hat/CentOS Live iso should just work by copying the original kernel cmdline parameters.

Could we have something similar for `mkosi`-generated  UKIs?

I know about [USI](https://overhead.neocities.org/blog/build-usi-mkosi/) but it's not exactly the same:
in my understanding, USI is ought to be loaded to RAM entirely, which is not necessary the case for a loopback device.
So USI is not very well suited for a full blown live Desktop Environment images, as Ubuntu installation ISO which is > 5Gb size.

**Comment by daandemeyer** [comment]
This is something to address in systemd, not mkosi

**Reply by bam80** [reply]
initrd where the loopback is handled is created by mkosi, isn't it?

**Reply by daandemeyer** [reply]
Yes and the goal of mkosi-initrd is to not implement features itself in the initrd but to delegate this to upstream projects such as systemd. So when systemd implements this functionality, we can expose it in mkosi.

**Reply by bam80** [reply]
https://github.com/systemd/systemd/issues/35604
